### DISCUSSION TITLE
seaborn.objects share x or y = False keeps all axes categorical

### DISCUSSION THREAD

**Discussion by subsurfaceiodev** [mwaskom/seaborn#3410] [Q&A]
Hi! As you may see, the x axis on the left subplot of the figure is displayed as categorical even though its data is numerical. Expected behavior would be for the x axis to be numerical and show the correct distance between x=2 and x=100.

```
import pandas as pd
import seaborn.objects as so

dataset = pd.DataFrame(dict(
    x=[1, 2, 10, 'A'],
    y=[1, 2, 2, 1],
    group=['g1', 'g1', 'g1', 'g2'],
    col=['c1', 'c1', 'c1', 'c2'],
))

p = (
    so.Plot(dataset,
            x='x',
            y='y',
            color='group',
            )
    .facet('col')
    .add(so.Dot())
    .share(x=False, y=True)
)
p.show()
```

![Figure_1](https://github.com/mwaskom/seaborn/assets/109676812/a2bb78ca-05dd-4fdc-86a8-7ebe06ec464e)

**Comment by frfeng** [comment]
Without looking at the source code, my guess is seaborn uses the data types from Pandas dtype (in this case x is an object rather than numerical because of the value `A`) and does not make further assumptions on the data type for each faceted data. Maybe there are reasons to not do that?

**Comment by mwaskom** [comment]
As [documented](https://seaborn.pydata.org/generated/seaborn.objects.Plot.share.html), `Plot.share` allows each facet to have different *axis limits* and *ticks*. However, the facets still share a *scale*, and so either need to treat the data as categorical or numerical, but not both. The facet operation show you slices of a variable, not distinct variables.

Depending on the nature of your data you may be able to use `Plot.pair`, which creates subplots for distinct variables.

**Comment by subsurfaceiodev** [comment]
We know seaborn is not meant to be an exact alternative to plotly express or viceversa, but the following code produces with px what one would expect:

```
pex = px.scatter(
    data_frame=dataset,
    x='x',
    y='y',
    color='group',
    facet_col='col'
)
```

![image](https://github.com/mwaskom/seaborn/assets/109676812/dc3d005f-108e-412c-badc-996ae3c3514a)
